jerry-libc/jerry-libc.c

```diff
@@ -176,7 +176,7 @@ CALL_PRAGMA (GCC diagnostic pop)
 
 /** Compare two strings. return an integer less than, equal to, or greater than zero
      if s1 is found, respectively, to be less than, to match, or be greater than s2.  */
-int
+int __attr_used___
 strcmp (const char *s1, const char *s2)
```

Comment:
This modification is unrelated to the -O3 issue. It's needed for this example, because compile of this code produces the following message: `'strcmp' referenced in section '.text' of <tmp object file> defined in discarded section '.text' of jerry-libc.c.o (symbol from plugin)`
Note: If the libraries and this example compiled without LTO, everything works fine.
I would like to provide a working example with default settings, so I would restore the solution with `strncmp`.
I agree with you, your example (after `> quitten = 3` the program will terminate) shows that this is not the best solution, but this is a simple example code. And force the compiler to keep this symbol at all costs is also not the best solution.
What do you think?